v1_0/api/metrics.py

```diff
@@ -0,0 +1,173 @@
+            'basic stats': [u'median number of downloads', u'average number of reads',
+                            u'normalized paper count', u'recent number of reads', u'number of papers',
+                            u'recent number of downloads', u'total number of reads',
+                            u'median number of reads', u'total number of downloads',
+                            u'average number of downloads'],
+            'basic stats refereed': [u'median number of downloads', u'average number of reads',
+                            u'normalized paper count', u'recent number of reads', u'number of papers',
+                            u'recent number of downloads', u'total number of reads',
+                            u'median number of reads', u'total number of downloads',
+                            u'average number of downloads'],
+            'citation stats': [u'normalized number of citations', u'average number of refereed citations',
+                               u'median number of citations', u'median number of refereed citations',
+                               u'number of citing papers', u'average number of citations',
+                               u'total number of refereed citations',
+                               u'normalized number of refereed citations',
+                               u'number of self-citations', u'total number of citations'],
+            'citation stats refereed': [u'normalized number of citations',
+                               u'average number of refereed citations',
+                               u'median number of citations', u'median number of refereed citations',
+                               u'number of citing papers', u'average number of citations',
+                               u'total number of refereed citations',
+                               u'normalized number of refereed citations',
+                               u'number of self-citations', u'total number of citations'],
+            'time series': [u'g', u'h', u'tori', u'i10', u'read10', u'i100']        
+        }
+        for entry in expected_stats:
+            self.assertItemsEqual(r.json()[entry].keys(), expected_stats[entry])
+        # There should be no skipped bibcodes
+        self.assertListEqual(r.json()['skipped bibcodes'], [])
+        # Sending an empty list of bibcodes to the service should give a 403
+        r = authenticated_user.post('/metrics', json={'bibcodes': []})
+        self.assertEqual(r.status_code, 403)
+        # Test getting metrics for just one bibcode via GET
+        r = authenticated_user.get('/metrics/%s' % bibcodes[0])
+        # We should get a 200
+        self.assertEqual(r.status_code, 200)
+        # And posting just one bibcode should work too
+        r = authenticated_user.post('/metrics', json={'bibcodes': bibcodes[:1]})
+        # We should get a 200 status
+        self.assertEqual(r.status_code, 200)
+    
+    def test_bumblebee_user(self):
```

Comment:
@ehenneken, hi Edwin,
do you think the authenticated user and the bumblebee user will diverge in the future? (in terms of access rights; access to different functionality as opposed to just having different max-limits)

if not, you could consider using one test for both, something along the lines of:

```
def test_authenticated_user(self, user=authenticated_user):
   r = user.post('metrics', json={'bibcodes': bibcodes})
   ....

def test_bumblebee_user(self)
    test_autenticated_user(user=bumblebee_user)
```